shuup/utils/money.py

```diff
@@ -60,3 +63,30 @@ def unit_matches_with(self, other):
 
     def new(self, value):
         return type(self)(value, currency=self.currency)
+
+    def as_rounded(self, digits=None):
+        """
+        Returns the value rounded to the given `digits` if specified,
+        or to the fetched `decimal_places` from the Currency
+        loaded from db using the `currency` attribute of this instance.
+
+        In case of non-existent currency in the db,
+        the fallback `digits` will always be 0.
+
+        :param digits int:
+          The number of digits to round
+
+        :return A new value rounded to the given digits
+        """
+
+        if digits is None:
+            # the digits info is not cached yet, fetch it
+            if self.currency not in digits_cache:
+                from shuup.core.models._currencies import Currency
+
+                currency = Currency.objects.filter(code=self.currency).first()
+                digits_cache[self.currency] = currency.decimal_places if currency else 0
+
+            return self.new(numbers.bankers_round(self.value, digits_cache[self.currency]))
+        else:
+            return self.new(numbers.bankers_round(self.value, digits))
```

Comment:
Instead of setting the precision provider. You could also use getter+settings combo like. Very least the app ready should set the precision provider from settings so it's easily changeable.